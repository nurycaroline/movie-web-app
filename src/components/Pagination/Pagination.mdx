---
name: Pagination
menu: Componentes
---

import {Playground} from 'docz';
import Pagination from '.';

# Pagination

<Playground>
  <Pagination
    pageActual="1"
    onClickBackPage={() => {}}
    onClickNextPage={() => {}}
    totalPages="26"
  />
</Playground>
