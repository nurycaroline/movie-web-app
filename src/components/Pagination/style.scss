@import '../../assets/variables';

.pagination {
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 15px;
  box-shadow: 0 0 5px $shadown-color;

  p {
    font-size: 16px;
  }

  button {
    background-color: $primary-color;
    padding: 10px 20px;
    font-size: 14px;
    color: $white;
    border: none;
    box-shadow: 0 0 5px $shadown-color;
    margin: 0 15px;

    &::disabled {
      background-color: $second-color;
    }
  }
}
