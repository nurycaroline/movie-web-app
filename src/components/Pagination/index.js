import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

const Pagination = ({
  pageActual,
  onClickBackPage,
  onClickNextPage,
  totalPages,
}) => (
  <div className="pagination">
    <button
      disabled={pageActual === 1}
      type="button"
      onClick={onClickBackPage}
    >
      Anterior
    </button>
    <p>
      {pageActual} de {totalPages}
    </p>
    <button type="button" onClick={onClickNextPage}>
      Proximo
    </button>
  </div>
);

Pagination.propTypes = {
  pageActual: PropTypes.number,
  onClickBackPage: PropTypes.func,
  onClickNextPage: PropTypes.func,
  totalPages: PropTypes.number,
};

Pagination.defaultProps = {
  pageActual: 1,
  onClickBackPage: () => {},
  onClickNextPage: () => {},
  totalPages: 0,
};

export default Pagination;
