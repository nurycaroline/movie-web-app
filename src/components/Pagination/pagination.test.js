import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from '.';

test('Pagination', () => {
  const component = renderer.create(<Pagination />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
