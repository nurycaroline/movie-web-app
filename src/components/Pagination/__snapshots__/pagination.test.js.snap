// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination 1`] = `
<div
  className="pagination"
>
  <button
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Anterior
  </button>
  <p>
    1
     de 
    0
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Proximo
  </button>
</div>
`;
